airflow/providers/databricks/sensors/table_changes.py

```diff
@@ -0,0 +1,152 @@
+        )
+        return sql_result
+
+    def _generate_query(
+        self,
+        prefix: str,
+        suffix: str,
+        table_name: str,
+        time_range_filter: str,
+    ) -> str:
+        formatted_opts = f"{prefix} {table_name}{suffix}{time_range_filter}"
+        self.log.debug("Formatted options: %s", formatted_opts)
+
+        return formatted_opts.strip()
+
+    @staticmethod
+    def get_previous_version(context: Context, lookup_key):
+        return context["ti"].xcom_pull(key=lookup_key, include_prior_dates=True)
+
+    @staticmethod
+    def set_version(context: Context, lookup_key, version):
+        context["ti"].xcom_push(key=lookup_key, value=version)
+
+    def get_current_table_version(self, table_name):
+        _prefix = "SELECT MAX(version) AS versions FROM (DESCRIBE HISTORY"
+        _data_operations_filter = ") WHERE operation NOT IN ('CONVERT', 'OPTIMIZE', 'CLONE', \
+        'RESTORE', 'FSCK') AND operation NOT LIKE 'VACUUM%'"
+        if self.timestamp is not None:
+            if self.change_filter_operator not in ("=", ">", "<", ">=", "<=", "!="):
+                raise AirflowException("Invalid comparison operator specified for time range filter.")
+            _timestamp_literal = f" AND timestamp {self.change_filter_operator} '{self.timestamp}'"
+        else:
+            _timestamp_literal = ""
+        query = self._generate_query(
+            prefix=_prefix,
+            suffix=_data_operations_filter,
+            time_range_filter=_timestamp_literal,
+            table_name=table_name,
+        )
+        self.log.debug("Query to be executed: %s", query)
+        result = self._sql_sensor(query)[0][0]
+        self.log.debug("Query result: %s", result)
+        return result
+
+    def _get_results_table_changes(self, context) -> bool:
+        complete_table_name = str(self.catalog + "." + self.schema + "." + self.table_name)
+        self.log.debug("Table name generated from arguments: %s", complete_table_name)
+
+        prev_version = -1
+        if context is not None:
+            lookup_key = complete_table_name
+            prev_data = self.get_previous_version(lookup_key=lookup_key, context=context)
+            self.log.debug("prev_data: %s, type=%s", str(prev_data), type(prev_data))
+            if isinstance(prev_data, int):
+                prev_version = prev_data
+            elif prev_data is not None:
+                raise AirflowException("Incorrect type for previous XCom data: %s", type(prev_data))
+            version = self.get_current_table_version(table_name=complete_table_name)
+            self.log.debug("Current version: %s", version)
+            if version is None:
+                return False
+            if prev_version < version:
+                result = True
+            else:
+                return False
+            if prev_version != version:
+                self.set_version(lookup_key=lookup_key, version=version, context=context)
```

Comment:
I don't think I fully understand this logic. The two False cases can certainly be collapsed to be more compact, but also, shouldn't the false case be setting `result` rather than returning? If they return then the code to store the version in xcom is not executed. You're basically always comparing `version` with the `prev_version` default value of `-1` from what I can tell.